Production: Deployment to Heroku failed

##### **Deployment to Heroku failed**
`       [INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5.jar (480 kB at 3.9 MB/s)
       [WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
       You are running an incompatible version of Java. JHipster supports JDK 8 to 12.
       [INFO] ------------------------------------------------------------------------
       [INFO] BUILD FAILURE
       [INFO] ------------------------------------------------------------------------
       [INFO] Total time:  03:05 min
       [INFO] Finished at: 2020-07-22T14:56:01Z
       [INFO] ------------------------------------------------------------------------
       [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce (enforce-versions) on project handled: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
       [ERROR] 
       [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
       [ERROR] Re-run Maven using the -X switch to enable full debug logging.
       [ERROR] 
       [ERROR] For more information about the errors and possible solutions, please read the following articles:
       [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
 !     ERROR: Failed to build app with Maven
       We're sorry this build is failing! If you can't find the issue in application code,
       please submit a ticket so we can help: https://help.heroku.com/
 !     Push rejected, failed to compile Java app.
 !     Push failed`

##### **Reproduce the error**

create app with default options, database postgresql.

-----> Java app detected
-----> Installing JDK 14... done
-----> Executing Maven
       $ ./mvnw -Pprod,heroku -DskipTests clean dependency:list install
       [INFO] Scanning for projects...
       [WARNING] 
       [WARNING] Some problems were encountered while building the effective model for com.gelowb:revertback:jar:0.0.1-SNAPSHOT
       [WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-checkstyle-plugin is missing. @ line 514, column 21
       [WARNING] 
       [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
       [WARNING] 
       [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
       [WARNING] 
       [INFO] 
       [INFO] -----------------------< com.gelowb:revertback >------------------------
       [INFO] Building Revertback 0.0.1-SNAPSHOT
       [INFO] --------------------------------[ jar ]---------------------------------
       [INFO] 
       [INFO] --- maven-clean-plugin:3.1.0:clean (default-clean) @ revertback ---
       [INFO] 
       [INFO] --- maven-dependency-plugin:2.8:list (default-cli) @ revertback ---
       [INFO] 
       [INFO] --- maven-resources-plugin:3.1.0:copy-resources (default-resources) @ revertback ---
       [INFO] Using 'UTF-8' encoding to copy filtered resources.
       [INFO] Copying 0 resource
       [INFO] Copying 21 resources
       [INFO] 
       [INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ revertback ---
       [INFO] Using 'UTF-8' encoding to copy filtered resources.
       [INFO] Copying 0 resource
       [INFO] Copying 21 resources
       [INFO] 
       [INFO] --- maven-enforcer-plugin:3.0.0-M1:enforce (enforce-versions) @ revertback ---
       [WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
       You are running an incompatible version of Java. JHipster supports JDK 8 to 12.
       [INFO] ------------------------------------------------------------------------
       [INFO] BUILD FAILURE
       [INFO] ------------------------------------------------------------------------
       [INFO] Total time:  4.896 s
       [INFO] Finished at: 2020-07-22T17:48:50Z
       [INFO] ------------------------------------------------------------------------
       [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M1:enforce (enforce-versions) on project revertback: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
       [ERROR] 
       [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
       [ERROR] Re-run Maven using the -X switch to enable full debug logging.
       [ERROR] 
       [ERROR] For more information about the errors and possible solutions, please read the following articles:
       [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
 !     ERROR: Failed to build app with Maven
       We're sorry this build is failing! If you can't find the issue in application code,
       please submit a ticket so we can help: https://help.heroku.com/
 !     Push rejected, failed to compile Java app.
 !     Push failed

--- a/generators/heroku/index.js
+++ b/generators/heroku/index.js
@@ -1,5 +1,6 @@
 /* eslint-disable consistent-return,max-classes-per-file */
 const chalk = require('chalk');
+const _ = require('lodash');
 const fs = require('fs');
 const ChildProcess = require('child_process');
 const util = require('util');
@@ -69,6 +70,7 @@ module.exports = class extends HerokuGeneratorOverride {
                 const configuration = this.getAllJhipsterConfig(this, true);
                 this.env.options.appPath = configuration.get('appPath') || constants.CLIENT_MAIN_SRC_DIR;
                 this.baseName = configuration.get('baseName');
+                this.dasherizedBaseName = _.kebabCase(this.baseName);
                 this.packageName = configuration.get('packageName');
                 this.packageFolder = configuration.get('packageFolder');
                 this.cacheProvider = configuration.get('cacheProvider') || configuration.get('hibernateCache') || 'no';

--- a/generators/heroku/templates/Procfile.ejs
+++ b/generators/heroku/templates/Procfile.ejs
@@ -16,7 +16,7 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-web: java $JAVA_OPTS <% if (applicationType === 'gateway' || dynoSize === 'Free') { %>-Xmx128m<% } %> -Dmicronaut.env.deduction=false -Dmicronaut.environments=prod,cloud<% if (buildTool == 'maven' && herokuDeployType == 'git') { %>,no-liquibase<% } %>,heroku -Dmicronaut.server.port=$port -jar <% if (buildTool === 'maven') { %>target/*.jar<% } %><% if (buildTool === 'gradle') { %>build/libs/*-all.jar<% } %>
+web: java $JAVA_OPTS <% if (applicationType === 'gateway' || dynoSize === 'Free') { %>-Xmx128m<% } %> -Dmicronaut.env.deduction=false -Dmicronaut.environments=prod,cloud<% if (buildTool == 'maven' && herokuDeployType == 'git') { %>,no-liquibase<% } %>,heroku -Dmicronaut.server.port=$port -jar <% if (buildTool === 'maven') { %>target/<%= dasherizedBaseName %>*.jar<% } %><% if (buildTool === 'gradle') { %>build/libs/*-all.jar<% } %>
 <%_ if (buildTool == 'maven' && herokuDeployType == 'git' && (prodDatabaseType === 'postgresql' || prodDatabaseType === 'mysql' || prodDatabaseType === 'mariadb')) { _%>
 release: cp -R src/main/resources/config config && ./mvnw -ntp liquibase:update -Pprod,heroku
 <%_ } _%>

